public/app/plugins/datasource/loki/responseUtils.ts

```diff
@@ -123,7 +125,38 @@ function shouldCombine(frame1: DataFrame, frame2: DataFrame): boolean {
     return false;
   }
 
-  return frame1.name === frame2.name;
+  const frameType1 = frame1.meta?.type;
+  const frameType2 = frame2.meta?.type;
+
+  if (frameType1 !== frameType2) {
+    // we do not join things that have a different type
+    return false;
+  }
+
+  // metric range query data
+  if (frameType1 === DataFrameType.TimeSeriesMulti) {
+    const field1 = frame1.fields.find((f) => f.type === FieldType.number);
+    const field2 = frame2.fields.find((f) => f.type === FieldType.number);
+    if (field1 === undefined || field2 === undefined) {
+      // should never happen
+      return false;
+    }
+
+    return shallowCompare(field1.labels ?? {}, field2.labels ?? {});
```

Comment:
Is this really enough to determine if two frames "belong to each other"?

In a test I did both fields were an empty object:
![image](https://user-images.githubusercontent.com/8092184/232501259-9fc8378c-80f2-4c7d-95c5-0141d837d739.png)

But even if they would be filled with the labels, how would two queries with the same labels we handled?

E.g. 
![image](https://user-images.githubusercontent.com/8092184/232501417-b2e7a9c2-be6b-4164-b44a-5494d4e59d20.png)
I would assume they have the same labels?

Reply:
Thanks @gabor. Sounds good to me. Regarding the missing labels on the Loki metric responses: are they missing because we haven't implemented them yet?

Reply:
Still doesn't produce any labels, but the feature itself is working.